plugins/woocommerce/includes/admin/class-wc-admin-menus.php

```diff
@@ -86,6 +86,13 @@ public function settings_menu() {
 		add_action( 'load-' . $settings_page, array( $this, 'settings_page_init' ) );
 	}
 
+	/**
+	 * Check if the user can access the top-level WooCommerce item.
+	 */
+	public static function can_view_woocommerce_menu_item() {
+		return current_user_can( 'edit_others_shop_orders' );
```

Comment:
Nice!  Looks good overall, but before we merge this in, I'm also wondering if a filter would work for our needs here.

```php
return apply_filters( 'woocommerce_menu_item_visibility', current_user_can( 'edit_others_shop_orders' ) );
```

Does this work for the use case in WooCommerce Admin or is a separate menu item altogether?

Reply:
Good call! I added the filter in commit 2bc2d54

> Does this work for the use case in WooCommerce Admin or is a separate menu item altogether?

I like the hook addition and it looks like it'll still work for our use case in WooCommerce Admin.

Reply:
After analyzing in detail with @joshuatf the addition of this filter, we decided to roll this change back.